File: Examples/VisualBasic/ModifyingAndConvertingImages/WebPImages/ExportToWebP.vb

Imports Aspose.Imaging.ImageOptions

'
'This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Imaging for .NET API reference 
'when the project is build. Please check https:// Docs.nuget.org/consume/nuget-faq for more information. 
'If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from http://www.aspose.com/downloads, 
'Install it and then add its reference to this project. For any issues, questions or suggestions 
'Please feel free to contact us using http://www.aspose.com/community/forums/default.aspx
'


Namespace Aspose.Imaging.Examples.VisualBasic.ModifyingAndConvertingImages.WebPImages
    Class ExportToWebP
        Public Shared Sub Run()
            ' ExStart:ExportToWebP
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_WebPImages()

            ' Create an instance of image class.
            Using image__1 As Image = Image.Load(dataDir & Convert.ToString("SampleImage1.bmp"))
                ' Create an instance of WebPOptions class
                Dim options As New WebPOptions()

                ' Set Quality
                options.Quality = 50

                ' Set LossLess property
                options.Lossless = False

                ' Save the image in WebP format.
                image__1.Save(dataDir & Convert.ToString("ExportToWebP_out.webp"), options)
            End Using
            ' ExEnd:ExportToWebP
        End Sub
    End Class
End Namespace

File: Examples/CSharp/ModifyingAndConvertingImages/WebPImages/ExportToWebP.cs
// GIST-ID: a0cd791be80598432be50ecdb85a04c8
using Aspose.Imaging;
using Aspose.Imaging.ImageOptions;
using System;

/*
This project uses the Automatic Package Restore feature of NuGet to resolve the Aspose.Imaging for .NET API reference
when the project is built. Please check https://learn.microsoft.com/en-us/nuget/resources/nuget-faq for more information.
If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from https://releases.aspose.com/,
install it, and then add its reference to this project. For any issues, questions, or suggestions,
please feel free to contact us using https://forum.aspose.com/.
*/

namespace Aspose.Imaging.Examples.CSharp.ModifyingAndConvertingImages.WebPImages
{
    class ExportToWebP
    {
        public static void Run()
        {
            Console.WriteLine("Running example ExportToWebP");
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_WebPImages();

            // Create an instance of the Image class.
            using (Image image = Image.Load(dataDir + "SampleImage1.bmp"))
            {
                // Create an instance of WebPOptions and set its properties.
                WebPOptions options = new WebPOptions
                {
                    Quality = 50,
                    Lossless = false
                };
                image.Save(dataDir + "ExportToWebP_out.webp", options);
            }

            Console.WriteLine("Finished example ExportToWebP");
        }
    }
}